tests/org/intellij/erlang/resolve/ErlangFunctionCallParameterResolutionTest.java

```diff
@@ -0,0 +1,72 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.intellij.erlang.resolve;
+
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiReference;
+import com.intellij.psi.ResolveResult;
+import com.intellij.psi.impl.source.resolve.reference.impl.PsiMultiReference;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.util.ObjectUtils;
+import org.intellij.erlang.psi.ErlangFunction;
+import org.intellij.erlang.psi.ErlangModule;
+import org.intellij.erlang.psi.ErlangQAtom;
+import org.intellij.erlang.utils.ErlangLightPlatformCodeInsightFixtureTestCase;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class ErlangFunctionCallParameterResolutionTest extends ErlangLightPlatformCodeInsightFixtureTestCase {
+  @Override
+  protected String getTestDataPath() {
+    return "testData/resolve/functionCallParameter/";
+  }
+
+  protected void doTest(@NotNull Class<? extends PsiElement> psiClass, String expectedModulePath, String... filePaths) {
+    myFixture.configureByFiles(filePaths);
+    PsiElement focusedElement = myFixture.getFile().findElementAt(myFixture.getEditor().getCaretModel().getOffset());
+    focusedElement = PsiTreeUtil.getParentOfType(focusedElement, ErlangQAtom.class);
+    assertNotNull(focusedElement);
+    PsiReference reference = focusedElement.getReference();
+    assertNotNull(reference);
+    PsiElement resolvedElement = resolveToPsiElement(reference, psiClass);
+    assertNotNull(resolvedElement);
+    PsiFile containingFile = resolvedElement.getContainingFile();
+    assertNotNull(containingFile);
+    VirtualFile virtualFile = containingFile.getVirtualFile();
+    assertNotNull(virtualFile);
+    String actualModulePath = virtualFile.getPath();
+    assertTrue(actualModulePath.endsWith(expectedModulePath));
+  }
+
+  @Nullable
+  private static PsiElement resolveToPsiElement(@NotNull PsiReference reference,
+                                                @NotNull Class<? extends PsiElement> psiClass) {
+    PsiMultiReference multiReference = ObjectUtils.tryCast(reference, PsiMultiReference.class);
+    if (multiReference != null) {
+      for (ResolveResult result : multiReference.multiResolve(true)) {
+        PsiElement element = ObjectUtils.tryCast(result.getElement(), psiClass);
+        if (element != null) return element;
+      }
+    }
+    return reference.resolve();
+  }
+
+  public void testModuleParameter()   { doTest(ErlangModule.class, "module.erl", "moduleTest.erl", "module.erl"); }
+  public void testFunctionParameter() { doTest(ErlangFunction.class, "module.erl", "functionTest.erl", "module.erl"); }
```

Comment:
It should not be the same code - as I've already explained, these tests you wrote are insufficient - they don't really test where you're resolved. Also, it will get much simpler - there will be no need to pass psi element's class around.

You can also extract some logic already written in doTest and reuse it.